resources/hooks/local/pre-push

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    break
+done
+    
+# Fetch the GIT diff and format it as command input:
+DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
```

Comment:
I believe up to now Grumphp works on pre-commit only? In that case the diff is fine, since you won't be deleting branches, or you won't be in detached HEAD state, and the status of the local and remote branches are not important.

Only when you do a pre-push then this becomes important and acting solely on a diff is not going to be enough.